pkg/service/targetgroupsvc/update.go

```diff
@@ -0,0 +1,58 @@
+package targetgroupsvc
+
+import (
+	"context"
+	"strings"
+
+	"github.com/common-fate/common-fate/pkg/storage"
+	"github.com/common-fate/common-fate/pkg/targetgroup"
+	"github.com/common-fate/common-fate/pkg/types"
+	"github.com/common-fate/ddb"
+	"go.uber.org/zap"
+)
+
+type UpdateOpts struct {
+	TargetGroupID string
+	UpdateRequest types.CreateTargetGroupRequest
+}
+
+func (s *Service) UpdateTargetGroup(ctx context.Context, req UpdateOpts) (*targetgroup.TargetGroup, error) {
+	log := zap.S()
+
+	//get target group for updating
+
+	q := storage.GetTargetGroup{ID: req.TargetGroupID}
+
+	_, err := s.DB.Query(ctx, &q)
+	if err != nil && err != ddb.ErrNoItems {
```

Comment:
This ddb.ErrNoItems needs to be handled.

One way to handle this is to pass the targetGroup in as an input to the update method.

You would fetch the targetGroup in the api layer and return a 404 error if its not found, and the service just validates the input and updates the record

Reply:
The update method can be removed, see above comment